bumble/l2cap.py

```diff
@@ -1491,7 +1500,49 @@ def deregister_fixed_channel(self, cid: int) -> None:
         if cid in self.fixed_channels:
             del self.fixed_channels[cid]
 
-    def register_server(self, psm: int, server: Callable[[Channel], Any]) -> int:
+    def register_server(
```

Comment:
similar comment as for the open/connect comment: having  `max_credits`, `mtu` and `mps` argument for non-connection non-credit channels is misleading. Having two functions would be clearer. Maybe we can name them `Device.register_l2cap_server` and `Device.register_l2cap_le_credit_based_server` to keep a naming convention consistent with `Device.connect_l2cap_le_credit_based_channel()`  and `Device.open_l2cap_channel()`.  At the `l2cap.py` level, we can drop the `l2cap` part of the name and have just `register_server` and `register_l2_credit_based_server`.